packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,1503 @@
+   * (https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html) in the
+   * Amazon VPC User Guide.
+   */
+  readonly subnets: ec2.ISubnet[];
+}
+
+/**
+ * The minimum TLS version required for traffic to the domain.
+ */
+export enum TLSSecurityPolicy {
+  /** Cipher suite TLS 1.0 */
+  TLS_1_0 = 'Policy-Min-TLS-1-0-2019-07',
+  /** Cipher suite TLS 1.2 */
+  TLS_1_2 = 'Policy-Min-TLS-1-2-2019-07'
+}
+
+/**
+ * Specifies options for fine-grained access control.
+ */
+export interface AdvancedSecurityOptions {
+  /**
+   * ARN for the master user. Only specify this or masterUserName, but not both.
+   *
+   * @default - fine-grained access control is disabled
+   */
+  readonly masterUserArn?: string;
+
+  /**
+   * Username for the master user. Only specify this or masterUserArn, but not both.
+   *
+   * @default - fine-grained access control is disabled
+   */
+  readonly masterUserName?: string;
+
+  /**
+   * Password for the master user.
+   *
+   * You can use `SecretValue.plainText` to specify a password in plain text or
+   * use `secretsmanager.Secret.fromSecretAttributes` to reference a secret in
+   * Secrets Manager.
+   *
+   * @default - A Secrets Manager generated password
+   */
+  readonly masterUserPasswordSecret?: cdk.SecretValue;
+}
+
+/**
+ * Properties for an AWS Elasticsearch Domain.
+ */
+export interface DomainProps {
+  /**
+   * Domain Access policies.
+   *
+   * @default - No access policies.
+   */
+  readonly accessPolicies?: iam.PolicyStatement[];
+
+  /**
+   * Additional options to specify for the Amazon ES domain.
+   *
+   * @default - no advanced options are specified
+   */
+  readonly advancedOptions?: { [key: string]: (string) };
+
+  /**
+   * Configures Amazon ES to use Amazon Cognito authentication for Kibana.
+   *
+   * @default - Cognito not used for authentication to Kibana.
+   */
+  readonly cognitoKibanaAuth?: CognitoOptions;
+
+  /**
+   * Enforces a particular physical domain name.
+   *
+   * @default - A name will be auto-generated.
+   */
+  readonly domainName?: string;
+
+  /**
+   * The configurations of Amazon Elastic Block Store (Amazon EBS) volumes that
+   * are attached to data nodes in the Amazon ES domain.
+   *
+   * @default - 10 GiB General Purpose (SSD) volumes per node.
+   */
+  readonly ebs?: EbsOptions;
+
+  /**
+   * The cluster capacity configuration for the Amazon ES domain.
+   *
+   * @default - 1 r5.large.elasticsearch data node; no dedicated master nodes.
+   */
+  readonly capacity?: CapacityConfig;
+
+  /**
+   * The cluster zone awareness configuration for the Amazon ES domain.
+   *
+   * @default - no zone awareness (1 AZ)
+   */
+  readonly zoneAwareness?: ZoneAwarenessConfig;
+
+  /**
+   * The Elasticsearch version that your domain will leverage.
+   *
+   * Per https://aws.amazon.com/elasticsearch-service/faqs/, Amazon Elasticsearch Service
+   * currently supports Elasticsearch versions 7.7, 7.4, 7.1, 6.8, 6.7, 6.5, 6.4, 6.3, 6.2,
```

Comment:
No need to mention this here, its bound to get outdated.

Reply:
I have remove the version numbers.